sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/DefaultOfferRequirementProvider.java

```diff
@@ -135,17 +135,30 @@ private TaskRequirement getNewTaskRequirement(
     private static Collection<ResourceRequirement> getResourceRequirements(
             TaskSpec taskSpec, Collection<Protos.Resource> resources) {
         ResourceSet resourceSet = taskSpec.getResourceSet();
+
         Map<String, Protos.Resource> resourceMap = resources == null ?
                 Collections.emptyMap() :
-                resources.stream().collect(Collectors.toMap(r -> r.getName(), Function.identity()));
+                resources.stream()
+                        .filter(resource -> !resource.hasDisk())
+                        .collect(Collectors.toMap(r -> r.getName(), Function.identity()));
+
+        Map<String, Protos.Resource> volumeMap = resources == null ?
+                Collections.emptyMap() :
+                resources.stream()
+                        .filter(resource -> resource.hasDisk())
+                        .filter(resource -> resource.getDisk().hasVolume())
```

Comment:
Nit: I think this can be merged with the previous filter:
`.filter(resource -> resource.hasDisk() && resource.getDisk().hasVolume())`